src/libraries/System.Net.Http/tests/UnitTests/Headers/HttpHeadersTest.cs

```diff
@@ -2638,6 +2638,49 @@ public void TryAddWithoutValidation_ThreeValidValueHeader_UseGenericImplementati
             Assert.True(expectedValues.SequenceEqual(values));
         }
 
+        [Theory]
+        [InlineData(new object[] { "chunked" }, true)]
+        [InlineData(new object[] { "gzip", "chunked" }, true)]
+        [InlineData(new object[] { "gzip", "chunked", "deflate", "chunked" }, true)]
+        [InlineData(new object[] { true }, true)]
+        [InlineData(new object[] { true, null, true }, true)]
+        [InlineData(new object[] { "gzip", true }, true)]
+        [InlineData(new object[] { "gzip", true, "deflate", true }, true)]
+        [InlineData(new object[] { true, "deflate", true }, true)]
+        [InlineData(new object[] { false, "deflate", true }, true)]
+        [InlineData(new object[] { false }, false)]
+        [InlineData(new object[] { false, null, false }, false)]
+        [InlineData(new object[] { "gzip", false }, false)]
+        [InlineData(new object[] { false, "gzip" }, false)]
+        [InlineData(new object[] { "gzip", true, false }, false)]
+        [InlineData(new object[] { "gzip", true, "deflate" }, false)]
+        [InlineData(new object[] { "gzip", true, "deflate", false }, false)]
+        [InlineData(new object[] { }, null)]
+        [InlineData(new object[] { null }, null)]
+        [InlineData(new object[] { "gzip" }, null)]
+        [InlineData(new object[] { "chunked", "gzip" }, null)]
+        [InlineData(new object[] { "gzip", "deflate" }, null)]
+        [InlineData(new object[] { "gzip", "chunked", "gzip" }, null)]
+        [InlineData(new object[] { "gzip", true, null}, null)]
+        public void AddTransferEncoding_SetValues_ChunkedAsExpected(object?[] values, bool? expected)
+        {
+            var headers = new HttpRequestHeaders();
```

Comment:
Shouldn't we also test `HttpResponseHeaders`?

Reply:
They both use same base class implementation, so although it would make sense to test both, with current state of code, it considered it redundant. I believe that change these two implementations will diverge in future as very small.
But yes, we can, with very small effort test both. Shall we?

Reply:
My personal preference is to test the entire public surface regardless of the implementation, but if the existing header tests don't do the same, it's better to remain consistent with those.

Reply:
I see lots of test isolateed to requests or response headers only, and on some cases the testing my make sense for only one or the other.
I will make changes to test both, consistency or not.